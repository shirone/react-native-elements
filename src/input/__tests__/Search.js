import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Search from '../Search';

describe('Search component', () => {
  it('should render without issues', () => {
    const component = shallow(<Search />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
