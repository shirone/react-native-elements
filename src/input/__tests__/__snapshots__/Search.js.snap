// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Array [
            Object {
              "backgroundColor": "#393e42",
              "borderBottomColor": "#000",
              "borderBottomWidth": 1,
              "borderTopColor": "#000",
              "borderTopWidth": 1,
            },
            false,
            undefined,
          ]
    }
>
    <TextInput
        placeholderTextColor="#86939e"
        selectionColor="#86939e"
        style={
            Array [
                Object {
                  "backgroundColor": "#303337",
                  "borderRadius": 3,
                  "color": "#86939e",
                  "fontSize": 17.5,
                  "height": 30,
                  "margin": 8,
                  "overflow": "hidden",
                  "paddingLeft": 26,
                  "paddingRight": 19,
                },
                false,
                false,
                false,
                undefined,
              ]
        }
        underlineColorAndroid="transparent"
    />
    <Icon
        allowFontScaling={false}
        color="#86939e"
        name="search"
        size={16}
        style={
            Array [
                Object {
                  "backgroundColor": "transparent",
                  "left": 16,
                  "position": "absolute",
                  "top": 15.5,
                },
                undefined,
              ]
        }
    />
</View>,
  "nodes": Array [
    <View
      style={
            Array [
                  Object {
                    "backgroundColor": "#393e42",
                    "borderBottomColor": "#000",
                    "borderBottomWidth": 1,
                    "borderTopColor": "#000",
                    "borderTopWidth": 1,
                  },
                  false,
                  undefined,
                ]
      }
>
      <TextInput
            placeholderTextColor="#86939e"
            selectionColor="#86939e"
            style={
                  Array [
                        Object {
                          "backgroundColor": "#303337",
                          "borderRadius": 3,
                          "color": "#86939e",
                          "fontSize": 17.5,
                          "height": 30,
                          "margin": 8,
                          "overflow": "hidden",
                          "paddingLeft": 26,
                          "paddingRight": 19,
                        },
                        false,
                        false,
                        false,
                        undefined,
                      ]
            }
            underlineColorAndroid="transparent"
      />
      <Icon
            allowFontScaling={false}
            color="#86939e"
            name="search"
            size={16}
            style={
                  Array [
                        Object {
                          "backgroundColor": "transparent",
                          "left": 16,
                          "position": "absolute",
                          "top": 15.5,
                        },
                        undefined,
                      ]
            }
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Search
        icon={Object {}}
        lightTheme={false}
        loadingIcon={Object {}}
        noIcon={false}
        placeholderTextColor="#86939e"
        round={false}
        showLoadingIcon={false}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Search {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "icon": Object {},
          "lightTheme": false,
          "loadingIcon": Object {},
          "noIcon": false,
          "placeholderTextColor": "#86939e",
          "round": false,
          "showLoadingIcon": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "#393e42",
                                "borderBottomColor": "#000",
                                "borderBottomWidth": 1,
                                "borderTopColor": "#000",
                                "borderTopWidth": 1,
                              },
                              false,
                              undefined,
                            ]
          }
>
          <TextInput
                    placeholderTextColor="#86939e"
                    selectionColor="#86939e"
                    style={
                              Array [
                                        Object {
                                          "backgroundColor": "#303337",
                                          "borderRadius": 3,
                                          "color": "#86939e",
                                          "fontSize": 17.5,
                                          "height": 30,
                                          "margin": 8,
                                          "overflow": "hidden",
                                          "paddingLeft": 26,
                                          "paddingRight": 19,
                                        },
                                        false,
                                        false,
                                        false,
                                        undefined,
                                      ]
                    }
                    underlineColorAndroid="transparent"
          />
          <Icon
                    allowFontScaling={false}
                    color="#86939e"
                    name="search"
                    size={16}
                    style={
                              Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                          "left": 16,
                                          "position": "absolute",
                                          "top": 15.5,
                                        },
                                        undefined,
                                      ]
                    }
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "#393e42",
                                "borderBottomColor": "#000",
                                "borderBottomWidth": 1,
                                "borderTopColor": "#000",
                                "borderTopWidth": 1,
                              },
                              false,
                              undefined,
                            ]
          }
>
          <TextInput
                    placeholderTextColor="#86939e"
                    selectionColor="#86939e"
                    style={
                              Array [
                                        Object {
                                          "backgroundColor": "#303337",
                                          "borderRadius": 3,
                                          "color": "#86939e",
                                          "fontSize": 17.5,
                                          "height": 30,
                                          "margin": 8,
                                          "overflow": "hidden",
                                          "paddingLeft": 26,
                                          "paddingRight": 19,
                                        },
                                        false,
                                        false,
                                        false,
                                        undefined,
                                      ]
                    }
                    underlineColorAndroid="transparent"
          />
          <Icon
                    allowFontScaling={false}
                    color="#86939e"
                    name="search"
                    size={16}
                    style={
                              Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                          "left": 16,
                                          "position": "absolute",
                                          "top": 15.5,
                                        },
                                        undefined,
                                      ]
                    }
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Search
    icon={Object {}}
    lightTheme={false}
    loadingIcon={Object {}}
    noIcon={false}
    placeholderTextColor="#86939e"
    round={false}
    showLoadingIcon={false}
/>,
}
`;
